### PR TITLE
OPCM: DeployOPChain additional testing checks.

### PR DESCRIPTION
**Pull request by blmalone**
Adding some additional checks `PermissionedDisputeGame` and `AnchorStateRegistryProxy`.

### CODE REVIEW

**Comment by blmalone** [comment]
* **#12355** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/12355?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @blmalone and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/contracts-bedrock/scripts/DeployOPChain.s.sol [hunk 1]
```diff
@@ -387,6 +387,10 @@ contract DeployOPChainOutput is BaseDeployIO {
         require(address(game.weth()) == address(delayedWETHPermissionedGameProxy()), "DPG-40");
         require(address(game.anchorStateRegistry()) == address(anchorStateRegistryProxy()), "DPG-50");
         require(game.l2ChainId() == _doi.l2ChainId(), "DPG-60");
+        require(game.l2BlockNumber() == 0, "DPG-70");
+        require(Duration.unwrap(game.clockExtension()) == 10800, "DPG-80");
```
Comment (mds1) [inline packages/contracts-bedrock/scripts/DeployOPChain.s.sol:391]: Should we add the `maxClockDuration()` check also?

File: packages/contracts-bedrock/test/opcm/DeployOPChain.t.sol [hunk 2]
```diff
@@ -544,6 +544,18 @@ contract DeployOPChain_Test is DeployOPChain_TestBase {
...
 
+        (Hash actualRoot,) = doo.anchorStateRegistryProxy().anchors(GameTypes.PERMISSIONED_CANNON);
+        assertEq(Hash.unwrap(actualRoot), 0xdead000000000000000000000000000000000000000000000000000000000000, "2900");
+        assertEq(doo.permissionedDisputeGame().l2BlockNumber(), 0, "3000");
+        assertEq(
+            Claim.unwrap(doo.permissionedDisputeGame().absolutePrestate()),
+            0x038512e02c4c3f7bdaec27d00edf55b7155e0905301e1a88083e4e0a6764d54c,
+            "3100"
+        );
+        assertEq(Duration.unwrap(doo.permissionedDisputeGame().clockExtension()), 10800, "3200");
```
Comment (mds1) [inline packages/contracts-bedrock/test/opcm/DeployOPChain.t.sol:555]: same question

**Review by mds1 — APPROVED** [review]
LGTM, just one missing check to add?

File: packages/contracts-bedrock/scripts/DeployOPChain.s.sol (hunk 1 above)
Comment (blmalone) [reply packages/contracts-bedrock/scripts/DeployOPChain.s.sol:391]: Good call: https://github.com/ethereum-optimism/optimism/pull/12355/commits/b8626321b210e7729b137a1633e14525ca3a5374

File: packages/contracts-bedrock/test/opcm/DeployOPChain.t.sol (hunk 2 above)
Comment (blmalone) [reply packages/contracts-bedrock/test/opcm/DeployOPChain.t.sol:555]: Added: https://github.com/ethereum-optimism/optimism/pull/12355/commits/b8626321b210e7729b137a1633e14525ca3a5374

**Pull request merged**